Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FlatPlatePaalmanPingsCorrection.py

```diff
@@ -656,35 +655,33 @@ def _container_transmission_calc(self, acc, acc1, acc2, ki_s, kf_s, ki_c1, kf_c2
 
     def _container_reflection_calc(self, acc, acc1, acc2, ki_s, kf_s, ki_c1, kf_c1, ass):
         if self._has_can_front_in and self._has_can_back_in:
-            acc = self._can_front_thickness * acc1
-            acc += self._can_back_thickness * acc2 * np.exp(-ki_c1 - kf_c1)
-            acc /= (self._can_front_thickness + self._can_back_thickness)
+            acc = (self._can_front_thickness * acc1 + self._can_back_thickness * acc2 * np.exp(-ki_c1 - kf_c1)) \
+                  / (self._can_front_thickness + self._can_back_thickness)
+            if self._has_sample_in:
+                acsc = (self._can_front_thickness * acc1 + self._can_back_thickness * acc2 * np.exp(-ki_c1 - ki_s - kf_s - kf_c1)) \
+                       / (self._can_front_thickness + self._can_back_thickness)
```

Comment:
Similar to the previous comment, a larger number of operations than necessary would be performed here. The (possible) solution could be the same as the previous one.